DscResources/MSFT_xSmbShare/MSFT_xSmbShare.psm1

```
@@ -1,3 +1,641 @@
+      
+      if($user){
+
+## If the User should should have permission we will  add him to this Variable
+
+      $TestingcorrectperrmissionAndparameters += $result
+      $addeduser = $PSBound.NoAccess
+      $AddingUserPermission.add($addeduser[$i],$result)
+
+      $ErrorActionPreference = "Continue"
+      
+      $user=$null
+      
+      }
+      
+     }
+        
+    }#end IF 
+
+If   ($PSBound.ContainsKey('readaccess')){
+
+     for ($i = 0; $i -lt $numberreadaccess; $i++)
+    { 
+      $ErrorActionPreference = "SilentlyContinue"
+
+     $result =  ($CurrentConfiguration.readaccess   -contains    $PSBound.readaccess[$i]).ToString() 
+
+     $user = $PSBound.readaccess[$i]
+     
+
+      if($user){
+
+
+## If the User should should have permission we will  add him to this Variable
+
+      $TestingcorrectperrmissionAndparameters += $result
+
+      $addeduser = $PSBound.readaccess
+
+      $ErrorActionPreference = "SilentlyContinue"
+      $AddingUserPermission.add($addeduser[$i],$result)
+
+      $user=$null
+
+      $ErrorActionPreference = "Continue"
+      } 
+
+     }   
+    
+    } #end IF
+
+If   ($PSBound.ContainsKey('fullaccess')){
+
+     for ($i = 0; $i -lt $numberfullaccess ; $i++)
+    { 
+     $ErrorActionPreference = "SilentlyContinue"
+
+     $result =  ($CurrentConfiguration.fullaccess  -contains    $PSBound.fullaccess[$i] ).ToString()  
+
+     $user = $PSBound.fullaccess[$i]
+     
+
+
+     if($user){
+
+
+## If the User should should have permission we will  add him to this Variable
+
+     $TestingcorrectperrmissionAndparameters += $result
+
+      $addeduser = $PSBound.fullaccess
+      $AddingUserPermission.add($addeduser[$i],$result)
+
+      $ErrorActionPreference = "Continue"
+      $user=$null
+
+      }
+
+        
+    }
+
+
+
+
+}#end IF
+
+If   ($PSBound.ContainsKey('ChangeAccess')){
+
+     for ($i = 0; $i -lt $numberChangeAccess; $i++)
+    { 
+     $ErrorActionPreference = "SilentlyContinue"
```

Why is silently continue needed?